src/library_exceptions.js

```diff
@@ -7,92 +7,132 @@
 var LibraryExceptions = {
   __exception_last: '0',
   __exception_caught: ' []',
-  __exception_infos: '{}',
-
-  __exception_deAdjust__deps: ['__exception_infos'],
-  __exception_deAdjust: function(adjusted) {
-    if (!adjusted || ___exception_infos[adjusted]) return adjusted;
-    for (var key in ___exception_infos) {
-      var ptr = +key; // the iteration key is a string, and if we throw this, it must be an integer as that is what we look for
-      var adj = ___exception_infos[ptr].adjusted;
-      var len = adj.length;
-      for (var i = 0; i < len; i++) {
-        if (adj[i] === adjusted) {
-#if EXCEPTION_DEBUG
-          err('de-adjusted exception ptr ' + adjusted + ' to ' + ptr);
-#endif
-          return ptr;
-        }
+
+  __ExceptionInfoAttrs: {
+    TYPE_OFFSET: 0,
+    DESTRUCTOR_OFFSET: Runtime.POINTER_SIZE,
+    REFCOUNT_OFFSET: Runtime.POINTER_SIZE * 2,
+    CAUGHT_OFFSET: Runtime.POINTER_SIZE * 2 + 4,
+    RETHROWN_OFFSET: Runtime.POINTER_SIZE * 2 + 5,
+
+    // Should be multiple of allocation alignment.
+    SIZE: alignMemory(Runtime.POINTER_SIZE * 2 + 6, 16)
+  },
```

Comment:
I wonder if this could actually be declared in C code and these offsets determined via the gen_struct_info process?

Perhaps we could have C implementations __cxa_free_exception, etc, since I'm not sure they make much sense in JS at all anymore.

Reply:
I agree C would be better. But I think what this PR does is very good in itself - it would solve an immediate problem, and also be simpler to convert to C later.

Reply:
Sure.. we can go conversion to C as a followup.